tool/tsh/kube.go

```diff
@@ -685,6 +741,12 @@ func (c *kubeCredentialsCommand) issueCert(cf *CLIConf) error {
 	})
 	span.End()
 	if err != nil {
+		var urlErr *url.Error
+		// If we've got network error we remove the lockfile, so we could restore from temporary connection
+		// problems without requiring user intervention.
+		if errors.As(err, &urlErr) {
+			deleteKubeCredsLock = true
+		}
```

Comment:
We should probably use `trace.IsConnectionProblem()` here. In any case, both are just heuristics, as you cannot know if the given error is persistent or not. 

In any case, shouldn't we delete the lock only after a period of time has passed? Otherwise, we can be back at the spam-tab problem.

Reply:
That's the error I'm getting when trying to run the code with network down, but I also added `trace.IsConnectionProblem()` check in there.
With network error with very high probability we don't enter the SSO login procedure at all, so we don't cause new tabs to be open and `tsh` quietly exits without side effects.